<a href="https://colab.research.google.com/github/comooney711/DSPS_CMooney/blob/main/Lab1/Copy_of_MLNPS_Ztestpandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
%pylab inline

Populating the interactive namespace from numpy and matplotlib


I extracted precipitations records for Willmington from 2000/01/01 through today. 
The goal is to extract from these data the daily precipitation starting on 2018/01/01 and assess if these data are different from a population with mean $\mu = 0.14$ standard deviation $\sigma= 0.38$. 

Set a p-value at $2\sigma$:  $p=0.05$

In [ ]:
rain = pd.read_csv("https://raw.githubusercontent.com/fedhere/DSPS/master/labs/1865331.csv")


The $Z$ statistics is : $Z = \frac{\bar{X} - \mu }{s}$ and $s = \sigma / \sqrt(N)$ (you should know what all those symbols mean)

1. Extract from the dataframe above the columns "DATE" and "PRCP" (precipitation)

2. Convert the dataframe DATE from ```string``` to ```datetime``` type
 using pd.datetime() 
    
3. Select the data rows with date later than 2018/01/01. 
    If your data type conversion worked well you should be able to set the condition as ```rain[<newcolumnname>]> pd.to_datetime("2018/01/01")```
    use the technique called "broadcasting" to select the rows: 
    
    ```rain[ rain[<newcolumnname>]> pd.to_datetime("2018/01/01")]``` which you can save as a new dataframe
    
4. Calculate the mean of the sample you created $\bar{X}$ and plug it in  the numerator of the $Z$ statistics: $\bar{X}- \mu$

5. Calculate the size of the sample and plug it in the formula. Now you have $Z$

6. Knowing the distribution of $Z\sim N(\mu=0, \sigma=1)$ and with a confidence $p=0.05$, did you find that the time period 2018/01/01---today has a statistically significant different amount of rain?  
    

In [ ]:
rain.head()

STATION                    NAME        DATE  ...  SNWD  WESD  WESF
0  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-01  ...   0.0   NaN   NaN
1  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-02  ...   NaN   NaN   NaN
2  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-03  ...   NaN   NaN   NaN
3  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-04  ...   NaN   NaN   NaN
4  US1DENC0010  NEWPORT 2.4 WNW, DE US  2010-01-05  ...   NaN   NaN   NaN

[5 rows x 8 columns]

In [ ]:
rain = rain[["DATE","PRCP"]]
rain.head()

DATE  PRCP
0  2010-01-01  0.28
1  2010-01-02  0.00
2  2010-01-03  0.00
3  2010-01-04  0.00
4  2010-01-05  0.00

In [ ]:
rain['DATE'] = pd.to_datetime(rain["DATE"] )

In [ ]:
rain2018 = rain[rain.DATE > pd.to_datetime("2018/01/01")]

In [ ]:
mu = 0.14
sigma = 0.38
xhat = rain2018['PRCP'].mean()
n = rain2018.shape[0]
Z = (xhat - mu)/ (sigma/ np.sqrt(n) )
Z

1.4686385901934103

From 01/01/2018 to today, the amount of rain is not statistically significant. This is because the Z statistic of my data is 1.47 sigma, which is less than a p value at 2 sigma. This corresponds to a p value of 0.07, which is greater than 0.05.